### PR TITLE
Add integration tests requiring secrets

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

**Comment by jdkandersson** [comment]
Just a note, I think @yanksyoon had been working on these kind of tests and may have experience with the issue you are seeing. Potentially worth waiting until next week when he is back, I do think these tests should be added back ASAP

File: .github/workflows/integration_test_secrets.yaml
Comment (gregory-schiano) [inline .github/workflows/integration_test_secrets.yaml:21]: Couldn't we reuse the image built by the other "integration" test ?
We'll be building and testing the images twice

File: .github/workflows/integration_test_secrets.yaml
Comment (gregory-schiano) [inline .github/workflows/integration_test_secrets.yaml:70]: This parameter is not used as openstack is not used (see comment above concerning devstack)

File: .github/workflows/integration_test_secrets.yaml
Comment (gregory-schiano) [inline .github/workflows/integration_test_secrets.yaml:76]: Same as above

File: .github/workflows/integration_test_secrets.yaml
Comment (gregory-schiano) [inline .github/workflows/integration_test_secrets.yaml:33]: Do we need to test with 1 and 3 units ?
I mean, is there a difference in the way secrets are managed between one and multiple units ?

File: .github/workflows/integration_test_secrets.yaml
Comment (arturo-seijas) [reply .github/workflows/integration_test_secrets.yaml:41]: Done

File: .github/workflows/integration_test.yaml
Comment (gregory-schiano) [inline .github/workflows/integration_test.yaml:39]: I suppose we'll wait for the branch `refactor-integration-test-job` to be merged before merging this one ?

File: tests/integration/test_core.py
Comment (gregory-schiano) [inline tests/integration/test_core.py:510]: Why do we skip this test ?

File: .github/workflows/integration_test_secrets.yaml
Comment (gregory-schiano) [reply .github/workflows/integration_test_secrets.yaml]: I saw your change, great, I close the comment here 👍

File: tests/integration/test_core.py
Comment (arturo-seijas) [reply tests/integration/test_core.py:510]: It seems like the launchpad web UI has changed and this test needs to read some elements in the forms in order to work. We'll probably need to rethink it moving forward to avoid flakiness. The purpose of this PR is to fix the CI set up, so I think we can leave this for the next PR.

File: .github/workflows/integration_test.yaml
Comment (arturo-seijas) [reply .github/workflows/integration_test.yaml:39]: Yes

File: .github/workflows/integration_test_secrets.yaml
Comment (arturo-seijas) [reply .github/workflows/integration_test_secrets.yaml:30]: AFAIK, none. But as part of this PR I'm only adapting what we already had. I lack context to change this but I see the same issue in some of the other integration tests.

File: .github/workflows/integration_test_secrets.yaml
Comment (arturo-seijas) [reply .github/workflows/integration_test_secrets.yaml]: https://github.com/canonical/operator-workflows/pull/90

**Review by amandahla — APPROVED** [review]
LGTM

**Review by gtrkiller — APPROVED** [review]
LGTM

**Comment by jdkandersson** [comment]
Thanks for adding them back and fixing the tests @arturo-seijas

**Pull request merged**
